CMakeLists.txt

```diff
@@ -302,9 +293,10 @@ elseif (APPLE)
   list(APPEND NANOGUI_EXTRA src/darwin.mm src/autorelease.mm)
   set_property(SOURCE src/autorelease.mm PROPERTY COMPILE_FLAGS -fno-objc-arc)
 elseif (CMAKE_SYSTEM MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "BSD")
-  list(APPEND NANOGUI_LIBS X11 pthread)
+  list(APPEND NANOGUI_LIBS pthread)
   if (NANOGUI_BACKEND STREQUAL "OpenGL")
-    list(APPEND NANOGUI_LIBS GL)
+    find_package(OpenGL REQUIRED COMPONENTS OpenGL)
+    list(APPEND NANOGUI_LIBS OpenGL::OpenGL)
```

Comment:
There's a possibility that this will not work with non-GLVND linux systems, though I can't tell how common these systems actually are. CMake's [FindOpenGL](https://cmake.org/cmake/help/latest/module/FindOpenGL.html) module doesn't provide clear guidance on the correct cross-platform solution here.